### DISCUSSION TITLE
Displayed error for GNSS/IMU coupling

### DISCUSSION THREAD

**Discussion by Adrienmangini** [rodralez/NaveGo#84] [Q&A] [answered]
Hello to all, 

I have an error when I run the ins_gnss function. 
Indeed, the error seems to appear in the file "att_update". The error is as follows: 
Error using att_update (line 73)
att_update: no attitude update mode defined.

Do you have an idea? 

Thank you very much!

**Comment by rodralez** [comment, marked as answer]
Hello,

The declaration of function `att_update` is:

`[qua, DCMbn, euler] = att_update(wb, DCMbn, qua, omega_ie_n, omega_en_n, dt, att_mode)`

Two attitude modes can be set: `dcm` and `quaternion`. If no `att_mode` is provided, the `quaternion` mode is the default attitude mode.

Maybe you misspelled the attitude mode, as `cuaternion` or `dcn`. In this case, no  attitude mode will be set.

Kind regards.

**Reply by Adrienmangini** [reply]
Thank you for your answer and your great work.

I am testing on my IMU and GNSS data carried by the human when running. 
 Do you think the coupling is feasible? 

Thank you very much in advance.

**Reply by rodralez** [reply]
Well, I have used navigation systems mounted on cars mostly.

Nevertheless, I think it is important for the runner to change heading frequently and run at a faster pace in order to make the Kalman filter states observable. You can check how many states are observable during the trajectory by analyzing the nav_e.ob variable.

It could be nice to know your results using NaveGo in human-mounted sensors.

Kind regards.

**Reply by Adrienmangini** [reply]
Thank you for your reactivity and your advice. 

I am working on the details. In the nav_e.obj variable, I get only 0's. In input, I don't know if I should enter particular values since, in my case, I use human.

**Reply by Adrienmangini** [reply]
Also, when I run "ins_gnss", it gives me Warning: pos_update: altitude is negative. 

Thank you in advance for your answer

**Reply by rodralez** [reply]
> Thank you for your reactivity and your advice.
> 
> I am working on the details. In the nav_e.obj variable, I get only 0's. In input, I don't know if I should enter particular values since, in my case, I use human.

Well, this means that all Kalman states are unobservable. This is a problem.

Generally, the main problem is the tuning or adjustment of the Kalman filter. In my opinion, this is the first problem you have to solve.

**Comment by hanegeyavuz** [comment]
> Also, when I run "ins_gnss", it gives me Warning: pos_update: altitude is negative.
> 
> Thank you in advance for your answer

did u solve the "altitude is negative" problem?
